department-of-veterans-affairs/caseflow
# Calvin/appeals 57225

Resolves [APPEALS-57225](https://jira.devops.va.gov/browse/APPEALS-57225)

# Description
Added methods from case_docket.rb to the aoj_case_docket, to ensure that when levers are infinite, AOJ appeals with hearings held are distributed correctly

Updated rspecs to include some new data

## Acceptance Criteria
*Value Statement*

As a developer, I need to adapt the existing CAVC bug fix to the AOJ docket appeals to ensure that appeals with a hearing held after remand are distributed correctly when the AOJ affinity lever is set to "Always distribute to current judge."

 
*Acceptance Criteria:*
 

- [ ]  When a user sets any of the AOJ affinity levers to "Always distribute to current judge" for AOJ docket appeals with a hearing held after remand, when requesting appeals it will distribute the appeal with an affinity to the deciding judge.

- [ ]  When a user sets any of the AOJ affinity levers to "Always distribute to current judge" for AOJ docket appeals with an ineligible hearing judge, the appeal will distribute with an affinity to the deciding judge, ensuring it is not distributed as genpop.

- [ ]  When a user sets any of the AOJ affinity levers to "Attempt to distribute" or "Omit," the appeal will distribute AOJ docket appeals according to the specified setting, with no impact from the "Always distribute to current judge" logic.

*Notes:*
 * Implementing bug fix in [APPEALS-57167](https://jira.devops.va.gov/browse/APPEALS-57167)

## Testing Plan

1.  Seed local with aoj data, `make clean && bundle exec rake db:seed:aoj_remand_return_legacy_appeals`
2. Set aoj_levers to infinite in case distro page
2. Login as an active judge user
3. Navigate to their judge assignment queue page
4. Request more cases
5. Verify that cases where the most recent hearing was held by an ineligible judge AND the previous decision was from an active judge AND are within their affinity window are not distributed to a judge who is not the previous deciding judge.

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
